File: scripts/maintenance/clean-session-stats.py
#!/usr/bin/env python3
import sys
import json
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from adapter.bridge.stats import cleanup
from adapter.bridge.context import hook_entrypoint

@hook_entrypoint(fallback_result={})
def main(context):
    if context.get('fullyIdle', False):
        conv_id = context.get('conversationId')
        if conv_id:
            cleanup(conv_id)
    return {}

if __name__ == "__main__":
    main()

File: packages/adapter-antigravity/src/maintenance/clean-session-stats.ts
import { cleanup } from "../bridge/stats";

export function main(context: Record<string, unknown>): Record<string, unknown> {
  try {
    return _main(context);
  } catch {
    return {};
  }
}

export function _main(context: Record<string, unknown>): Record<string, unknown> {
  if (context["fullyIdle"]) {
    const convId = context["conversationId"] as string | undefined;
    if (convId) cleanup(convId);
  }
  return {};
}

import { hookEntrypoint } from "../bridge/context";

if (typeof require !== "undefined" && require.main === module) {
  hookEntrypoint()(main)();
}

